### PR TITLE
Create 3,2-CSP sctructure and constructor

### PR DESCRIPTION
**Pull request by H3w9VH3F4L4Tq5MbsfRi**

### CODE REVIEW

File: 3-coloring-algorithm/32CSP.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using QuikGraph;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (Sosssen) [inline 3-coloring-algorithm/32CSP.cs:6]: to remove

File: 3-coloring-algorithm/32CSP.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            g = new();
+
+            for (int i = 0; i < g.VertexCount * A; i++)
+                g.AddVertex(i);
+
+            for (int i = 0; i < g.VertexCount; i++)
+            {
+                var vertices = g.AdjacentVertices(i);
+
+                foreach (var v in vertices)
```
Comment (Sosssen) [inline 3-coloring-algorithm/32CSP.cs:26]: {}
https://armannotes.com/2020/11/23/why-you-should-always-use-curly-braces-for-single-statement-blocks-in-csharp/

File: 3-coloring-algorithm/32CSP.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    internal class _32CSP
+    {
+        private UndirectedGraph<int, SUndirectedEdge<int>> g;
+        static readonly int A = 3;
+
+        public _32CSP(UndirectedGraph<int, SUndirectedEdge<int>> graph)
+        {
+            g = new();
+
+            for (int i = 0; i < g.VertexCount * A; i++)
```
Comment (Sosssen) [inline 3-coloring-algorithm/32CSP.cs:19]: {}
https://armannotes.com/2020/11/23/why-you-should-always-use-curly-braces-for-single-statement-blocks-in-csharp/

File: 3-coloring-algorithm/32CSP.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace _3_coloring_algorithm
+{
+    internal class _32CSP
+    {
+        private UndirectedGraph<int, SUndirectedEdge<int>> g;
+        static readonly int A = 3;
```
Comment (Sosssen) [inline 3-coloring-algorithm/32CSP.cs:13]: maybe rename to `a`? as in (a,b)-CSP

**Review by Sosssen — CHANGES_REQUESTED** [review]

File: 3-coloring-algorithm/32CSP.cs (hunk 4 above)
Comment (H3w9VH3F4L4Tq5MbsfRi) [reply 3-coloring-algorithm/32CSP.cs:13]: done

File: 3-coloring-algorithm/32CSP.cs (hunk 2 above)
Comment (H3w9VH3F4L4Tq5MbsfRi) [reply 3-coloring-algorithm/32CSP.cs:26]: done

File: 3-coloring-algorithm/32CSP.cs (hunk 1 above)
Comment (H3w9VH3F4L4Tq5MbsfRi) [reply 3-coloring-algorithm/32CSP.cs:6]: done

**Review by Sosssen — APPROVED** [review]
jakie smaczne

**Pull request merged**